app/services/orphaned_links_remover.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Service for removing orphaned links
+class OrphanedLinksRemover
+  class << self
+    # rubocop:disable Metrics/AbcSize
+    def run!
+      ProfileRule.left_outer_joins(:profile).left_outer_joins(:rule).where(profiles: { id: nil }).or(
+        ProfileRule.where(rules: { id: nil })
+      ).destroy_all
+
+      RuleResult.left_outer_joins(:rule).left_outer_joins(:host).where(rules: { id: nil }).or(
```

Comment:
I'm worried about the `.left_outer_joins(:host)` appending hosts to the RuleResuls and TestResults. Do we know how this would behave with production-grade data set?

Reply:
I can show you the output of the gabi query after lunch

Reply:
But what's wrong with the join?